File: travel_expense_app/lib/features/reimbursements/domain/entities/reimbursement_entity.dart
import 'package:equatable/equatable.dart';
import '../../../expenses/domain/entities/expense_claim_entity.dart';

class ReimbursementEntity extends Equatable {
  final String id;
  final String claimId;
  final ExpenseClaimEntity? claim;
  final String? paymentReference;
  final String status;
  final DateTime? paidAt;
  final DateTime createdAt;

  const ReimbursementEntity({
    required this.id,
    required this.claimId,
    this.claim,
    this.paymentReference,
    required this.status,
    this.paidAt,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, claimId, claim, paymentReference, status, paidAt, createdAt];
}

File: travel_expense_backend/src/modules/reimbursements/entities/reimbursement.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, OneToOne, JoinColumn } from 'typeorm';
import { ExpenseClaim } from '../../expenses/entities/expense-claim.entity';

export enum ReimbursementStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  PAID = 'paid',
  FAILED = 'failed',
}

@Entity('reimbursements')
export class Reimbursement {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  claimId: string;

  @OneToOne(() => ExpenseClaim, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'claimId' })
  claim: ExpenseClaim;

  @Column({ nullable: true })
  paymentReference: string;

  @Column({
    type: process.env.OFFLINE_MODE === 'true' ? 'simple-enum' : 'enum',
    enum: ReimbursementStatus,
    default: ReimbursementStatus.PENDING,
  })
  status: ReimbursementStatus;

  @Column({ nullable: true })
  paidAt: Date;

  @CreateDateColumn()
  createdAt: Date;
}
